.gitignore

```diff
@@ -12,6 +12,8 @@
 !/.markdownlintignore
 !/.pre-commit-config.yaml
 !/docs/CHANGELOG.md
+!/.github/sbom_metadata/metadata.cyclonedx.json
+!/.github/sbom_metadata/metadata.spdx.json
```

Comment:
These should go in `.github/.gitignore` as

```
!sbom_metadata/metadata.cyclonedx.json
!sbom_metadata/metadata.spdx.json
```

Reply:
This is mostly based on personal experience and opinion, but the original goal was:

- The project level .gitignore uses the allow list template, and only includes patterns anchored to the project root or found anywhere in the project
- The nested .gitignore file(s) are for whitelisting file patterns scoped to that directory and its subdirectories. It keeps everything modular, scoped, and tidy
- A single top level .gitignore can become very messy very quickly, and the patterns tend to become complicated due to the limited expression syntax of file globs
- Since everything is ignored except what is explicitly allowed, it's meant to reduce the chance of developers committing things they didn't intend to. There are of course things like pre-commit for this but not everyone uses it

I use similar patterns for things like Makefiles to keep everything contained to its enclosing scope. It's just the way I like to organize projects to keep my workspace clean (I typically have a lot of untracked files for testing or ideas I want to hold onto for later). If the overall consensus is a preference to use a single .gitignore file, I won't fight it, but I do think we need to do only one way or the other.

Reply:
@eddiezane @ashearin do you have any preferences on this?

Reply:
My preference for this repo would be a single `.gitignore`. Mostly for visibility for users and ease of maintenance: at least in a sense of intuitiveness for contributors. Just my opinion so I could be off base. I do agree that it could become messy though, something to watch. 

I could see a better argument for multiple if this was going the monorepo route, but I don't think thats likely. My feeling aren't strong, and I see advantages for both, so I'm good either way.

Reply:
@idunbarh here's one example

https://github.com/CycloneDX/specification/blob/master/docgen/json/.gitignore

Reply:
Sounds like the single top-level .gitignore gets the votes. @idunbarh do you want to make that change in this PR? I think this is all of them:

- internal/pkg/db/testdata/.gitignore
- .github/.gitignore
- .vscode/.gitignore